scripts/tests_like_ci/rust_tests.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+WD=$(git rev-parse --show-toplevel)
+
+# INPUTS
+
+# Running all the tests will compile different sets of crates and take a lot of storage (>500GB)
+# If your machine has less storage, you can run only part of the tests (at a time), 
+# use the name of the function to run as a subcommand, for instance:
+# ./scripts/tests_like_ci/rust_tests.sh simtests
+# the possible steps are: check_unused_deps, test_rust_crates, test_external_crates, test_extra, tests_using_postgres, simtests
+# the tests that need postgres will automatically (re-)start it
+RUN_ONLY_STEP=$1
+
+# restart postgres
+function restart_postgres() {
+    if ! command -v psql &> /dev/null; then
+        echo "'psql' is not installed in PATH. Please ensure it is installed and available."
+        exit 1
+    fi
+    docker rm -f -v $(docker ps -a | grep postgres | awk '{print $1}')
+    export POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-postgrespw}
+    export POSTGRES_USER=${POSTGRES_USER:-postgres}
+    export POSTGRES_DB=${POSTGRES_DB:-iota_indexer}
+    export POSTGRES_HOST=${POSTGRES_HOST:-postgres}
+    # assuming you run the indexer's postgres using docker-compose
+    cd ${WD}/docker/pg-services-local; docker-compose down -v postgres; docker-compose up -d postgres
+    PGPASSWORD=$POSTGRES_PASSWORD psql -h localhost -U $POSTGRES_USER -c 'CREATE DATABASE IF NOT EXISTS iota_indexer;' -c 'ALTER SYSTEM SET max_connections = 500;' 2>/dev/null
+}
+
+function retry_failing_only() {
+    filterset=""
+    for line in "${FAILING_NONSIM_TESTS[@]}"; do
+        arr=(${line// / })
+        if [ ${#arr[@]} -eq 2 ]; then
+            package=${arr[0]%%::*}
+            test_name=${arr[-1]#*::}
+            echo "package:$package test_name:$test_name"
+            filterset="${filterset} -E 'test(${test_name})'"
+            break   
+        fi
+    done
+    echo "FILTERSET: ${filterset}"
+    command="cargo nextest run --profile ci ${filterset} --test-threads 1"
+    set -x
+    eval $command
+}
+
+function test_rust_crates() {
+    # Tests written with #[sim_test] are often flaky if run as #[tokio::test] - this var
+    # causes #[sim_test] to only run under the deterministic `simtest` job, and not the
+    # non-deterministic `test` job.
+    export IOTA_SKIP_SIMTESTS=1
+    cargo nextest run --config-file .config/nextest.toml --profile ci
+}
+
+function test_external_crates() {
+    cargo nextest run  --config-file .config/nextest.toml   --manifest-path external-crates/move/Cargo.toml   -E '!test(prove) and !test(run_all::simple_build_with_docs/args.txt) and !test(run_test::nested_deps_bad_parent/Move.toml)'      --profile ci
```

Comment:
```suggestion
    cargo nextest run --config-file .config/nextest.toml --manifest-path external-crates/move/Cargo.toml -E '!test(prove) and !test(run_all::simple_build_with_docs/args.txt) and !test(run_test::nested_deps_bad_parent/Move.toml)' --profile ci
```
Could we also split these long lines up?

Reply:
sure thing !